### PR TITLE
ZCS-15053: Shared folder synchronization to mobile

### PR DESCRIPTION
**Pull request by miteshsavani810**
- Display sync to mobile checkbox on share which i have accepted section.
- Sync to mobile checkbox enable only if user have Admin or Manager permission and if it is contact, message or calendar share.

### CODE REVIEW

File: WebRoot/js/zimbraMail/prefs/view/ZmSharingPage.js [hunk 1]
```diff
@@ -835,11 +891,37 @@ function(html, idx, item, field, colIdx, params) {
...
 	}
 
 	return (params && params.returnText) ? html.join("") : idx;
 };
 
+ZmSharingListView.prototype._emulateSingleClick =
+function(params) {
+	if (
+		this.status === ZmSharingView.MOUNTED &&
+		params.ctrlKey &&
```
Comment (k-kato) [inline WebRoot/js/zimbraMail/prefs/view/ZmSharingPage.js:910]: Just a question. What value is expected in `params.ctrlKey`? boolean?
Is this param related to the following comment? If so, how is backquote considered?
```
handle selection of mobile sync checkbox using keyboard shortcut Ctrl + `.
```

File: WebRoot/js/zimbraMail/share/model/ZmSettings.js [hunk 2]
```diff
@@ -769,7 +769,6 @@ function(list, callback, result) {
 	if (resp.ModifyPrefsResponse != null) {
 		// notify each changed setting's listeners
 		for (var i = 0; i < list.length; i++) {
-			var setting = list[i];
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/share/model/ZmSettings.js:772]: removed by mistake?

Comment (k-kato) [inline WebRoot/js/zimbraMail/share/model/ZmSettings.js:772]: Doesn't this removal cause an error at the next line as `setting` is `undefined`.
```
undefined.origValue = undefined.copyValue();
```

File: WebRoot/messages/ZmMsg_ja.properties [hunk 3]
```diff
@@ -4493,4 +4494,6 @@ toolbar = \u30c4\u30fc\u30eb\u30d0\u30fc
 checked = \u30c1\u30a7\u30c3\u30af\u3057\u305f
 unChecked = \u672a\u30c1\u30a7\u30c3\u30af
 nextYear = \u6765\u5e74
-previousYear = \u524d\u5e74
\ No newline at end of file
+previousYear = \u524d\u5e74
+syncingToMobileBasedOnRole = \u6ce8: \u30e2\u30d0\u30a4\u30eb\u3078\u306e\u540c\u671f\u306b\u306f\u7ba1\u7406\u8005\u307e\u305f\u306f\u30de\u30cd\u30fc\u30b8\u30e3\u30fc\u306e\u5f79\u200b\u200b\u5272\u304c\u5fc5\u8981\u3067\u3042\u308a\u3001\u30b5\u30d6\u30d5\u30a9\u30eb\u30c0\u30fc\u304c\u3042\u308b\u5834\u5408\u306f\u305d\u308c\u3082\u542b\u307e\u308c\u307e\u3059\u3002
```
Comment (k-kato) [inline WebRoot/messages/ZmMsg_ja.properties:4498]: Are the two `U+200B` zero width space intended?
If it is not clear, please attach screenshots with and without them.
If they are unnecessary, please remove them.
```suggestion
syncingToMobileBasedOnRole = \u6ce8: \u30e2\u30d0\u30a4\u30eb\u3078\u306e\u540c\u671f\u306b\u306f\u7ba1\u7406\u8005\u307e\u305f\u306f\u30de\u30cd\u30fc\u30b8\u30e3\u30fc\u306e\u5f79\u5272\u304c\u5fc5\u8981\u3067\u3042\u308a\u3001\u30b5\u30d6\u30d5\u30a9\u30eb\u30c0\u30fc\u304c\u3042\u308b\u5834\u5408\u306f\u305d\u308c\u3082\u542b\u307e\u308c\u307e\u3059\u3002
```

File: WebRoot/js/zimbraMail/share/model/ZmSettings.js (hunk 2 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/share/model/ZmSettings.js]: Removed it by mistakenly. Revert the changes.

File: WebRoot/js/zimbraMail/prefs/view/ZmSharingPage.js (hunk 1 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/prefs/view/ZmSharingPage.js:910]: The value of `params.ctrlKey` is boolean. it will be true when user select the list item using keyboard shortcut  **Ctrl + backtick**.

The `_emulateSingleClick` is going to be called when user navigate to next/prev line using keyboard "Up/Down" arrow keys and selecting the list item using **Ctrl + backtick**.  

here we need tick/untick Sync to mobile checkbox when user perform **Ctrl + backtick** only hence we are just checking with params.ctrlKey value.

Comment (k-kato) [reply WebRoot/js/zimbraMail/prefs/view/ZmSharingPage.js:910]: I understand. Thanks!

**Review by k-kato — APPROVED** [review]

**Review by silentsakky — APPROVED** [review]

**Review by damini-vashishtha — APPROVED** [review]

**Review by silentsakky — APPROVED** [review]

**Review by lavegupta — APPROVED** [review]

**Review by damini-vashishtha — APPROVED** [review]

**Review by sitaram-mulik-synacor — APPROVED** [review]
